es/09_4_Codificando_una_P2PKH.md

```diff
@@ -0,0 +1,390 @@
+```
+Con eso en conjunto, puede examinar cómo se desbloquea un UTXO P2PKH.
+
+Primero, pone las constantes iniciales en la pila, y luego hace un duplicado de la pubKey con `OP_DUP`:
+```
+Script: <signature> <pubKey> OP_DUP OP_HASH160 <pubKeyHash> OP_EQUALVERIFY OP_CHECKSIG
+Stack: [ ]
+
+Script: <pubKey> OP_DUP OP_HASH160 <pubKeyHash> OP_EQUALVERIFY OP_CHECKSIG
+Stack: [ <signature> ]
+
+Script: OP_DUP OP_HASH160 <pubKeyHash> OP_EQUALVERIFY OP_CHECKSIG
+Stack: [ <signature> <pubKey> ]
+
+Script: OP_HASH160 <pubKeyHash> OP_EQUALVERIFY OP_CHECKSIG
+Running: <pubKey> OP_DUP
+Stack: [ <signature> <pubKey> <pubKey> ]
+```
+Por qué el duplicado? Porque es necesario comprobar los dos elementos de desbloqueo: la clave pública y la firma.
+
+A continuación, `OP_HASH160` saca el resultado de la pila `<pubKey>`, lo codifica y vuelve a colocar el resultado en la pila.
+```
+Script: <pubKeyHash> OP_EQUALVERIFY OP_CHECKSIG
+Running: <pubKey> OP_HASH160
+Stack: [ <signature> <pubKey> <pubKeyHash> ]
+```
+Luego, coloca el `<pubKeyHash>` que estaba en el script de bloqueo en la pila:
+```
+Script: OP_EQUALVERIFY OP_CHECKSIG
+Stack: [ <signature> <pubKey> <pubKeyHash> <pubKeyHash> ]
+```
+`OP_EQUALVERIFY` es efectivamente dos códigos de operación: `OP_EQUAL`, que saca dos elementos de la pila y empuja `True` o `False` en función de la comparación y `OP_VERIFY` que muestra ese resultado e inmediatamente marca la transacción como inválida si es `False`. (El capítulo 12 habla más sobre el uso de `OP_VERIFY` como condicional.)
+
+Suponiendo que los dos `<pubKeyHash>es` son iguales, obtendrá el siguiente resultado:
+```
+Script: OP_CHECKSIG
+Running: <pubKeyHash> <pubKeyHash> OP_EQUALVERIFY
+Stack: [ <signature> <pubKey> ]
+```
+En este punto, ha demostrado que el `<pubKey>` provisto en el `scriptSig` hashea a la dirección de Bitcoin en cuestión, por lo que sabe que el receptor conocía la clave pública. Pero también necesita demostrar el conocimiento de la clave privada, lo cual se hace con `OP_CHECKSIG`, lo que confirma que la firma del script de desbloqueo coincide con esa clave pública
+```
+Script:
+Running: <signature> <pubKey> OP_CHECKSIG
+Stack: [ True ]
+```
+El script ahora finaliza y, si tuvo éxito, la transacción puede volver a pasar el UTXO en cuestión.
+
+### Usando btcdeb para un ejemplo P2PKH
+
+Probar transacciones reales de Bitcoin con `btcdeb` es un poco más complicado, porque necesita conocer la clave pública y la firma para que todo funcione, y generar esta última es algo difícil. Sin embargo, una forma de probar las cosas es dejar que Bitcoin haga el trabajo por usted generando una transacción que _desbloquearía_ un UTXO. Eso es lo que ha hecho anteriormente: generar la transacción para gastar el UTXO hizo que `bitcoin-cli` calcule la` <firma> `y la` <pubKey> `. Luego, mira la información de transacción sin procesar del UTXO para aprender el script de bloqueo, incluido el `<pubKeyHash>`
```

Comment:
Correct spacing in `<firma>` and `<pubKey>`: ``generar la transacción para gastar el UTXO hizo que `bitcoin-cli` calcule la `<firma> `y la `<pubKey> `.``